x-pack/plugins/apm/public/context/LocationContext.tsx

```diff
@@ -4,28 +4,17 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { History, Location } from 'history';
-import React, { createContext, useState, useEffect } from 'react';
-
-interface Props {
-  history: History;
-}
+import { Location } from 'history';
+import React, { createContext } from 'react';
+import { withRouter } from 'react-router-dom';
 
 const initialLocation = {} as Location;
 
 const LocationContext = createContext(initialLocation);
-const LocationProvider: React.FC<Props> = ({ history, children }) => {
-  const [location, setLocation] = useState(history.location);
-
-  useEffect(() => {
-    const unlisten = history.listen(updatedLocation => {
-      setLocation(updatedLocation);
-    });
-
-    return unlisten;
-  }, []);
-
-  return <LocationContext.Provider children={children} value={location} />;
-};
+const LocationProvider: React.ComponentClass<{}> = withRouter(
+  ({ location, children }) => {
```

Comment:
Since we now have access to react-router we could also pass `matches` along. That would be very useful for `resolveUrlParams` so we can get rid of `getPathParams`

Reply:
@sqren not a react-router expert, but it seemingly only works when the component that uses the `withRouter` HOC is below a `Route`. The `LocationProvider` is below a `Router`, but not a `Route`, so `match` returns an object with a bunch of default (useless) values.

Reply:
> The LocationProvider is below a Router, but not a Route, so match returns an object with a bunch of default (useless) values.

Ahh, that rings a bell. Well, nvm then.